Description: allow pragma Assert with No_Implementation_Pragmas restriction.
 TODO:
 Upstream bug tracker says that the problem is fixed elsewhere since
 2015, so this patch is disabled. Check with the reproducer from
 upstream bug tracker, then remove this file.
Author: Eugeniy Meshcheryakov <eugen@debian.org>
Forwarded: https://gcc.gnu.org/bugzilla/show_bug.cgi?id=47818

--- a/src/gcc/ada/sem_prag.adb
+++ b/src/gcc/ada/sem_prag.adb
@@ -18629,7 +18629,16 @@ package body Sem_Prag is
             Typ_Arg : Node_Id;
 
          begin
-            GNAT_Pragma;
+            --  This could be a rewritten pragma Assert. If it is the case
+            --  then don't check restrictions, because they are different for
+            --  pragma Assert and were already checked.
+
+            if Nkind (Original_Node (N)) /= N_Pragma
+              or else Pragma_Name (Original_Node (N)) /= Name_Assert
+            then
+               GNAT_Pragma;
+            end if;
+
             Check_At_Least_N_Arguments (2);
             Check_At_Most_N_Arguments  (3);
             Check_Optional_Identifier (Arg1, Name_Entity);
